File: client/src/app/model/coin.ts
export interface Coin {
    id: string
    symbol: string
    name: string
    image: string
    current_price: number
    market_cap: number
    market_cap_rank: number
    fully_diluted_valuation: number
    total_volume: number
    high_24h: number
    low_24h: number
    price_change_24h: number
    price_change_percentage_24h: number
    market_cap_change_24h: number
    market_cap_change_percentage_24h: number
    circulating_supply: number
    total_supply: number
    max_supply: number
    ath: number
    ath_change_percentage: number
    ath_date: Date
    atl: number
    atl_change_percentage: number
    atl_date: string
    roi: Date
    last_updated: string
}

export interface SingleCoin {
    image: string
    name: string
    description: string
    marketCapRank: number
    marketDataCurrentPrice: number
    marketDataMarketCap: number
}

export const chartDays = [
  {
    label: '24 Hours',
    value: '1',
  },
  {
    label: '30 Days',
    value: '30',
  },
  {
    label: '3 Months',
    value: '90',
  },
  {
    label: '1 Year',
    value: '365',
  },
];

export class CoinPaigingParams {
  pageNumber: number;

  pageSize: number;

  constructor(pageNumber = 1, pageSize = 10) {
    this.pageNumber = pageNumber;
    this.pageSize = pageSize;
  }
}

export class ChartParmas {
  id: string;

  days: string;

  constructor(id = 'bitcoin', days = '1') {
    this.id = id;
    this.days = days;
  }
}

File: API/Entities/Coin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace API.Entities
{
    public class Coin
    {

        [JsonProperty("id")]
        public string Id { get; set; }
        [JsonProperty("symbol")]
        public string Symbol { get; set; }
        [JsonProperty("name")]
        public string Name { get; set; }
        [JsonProperty("image")]
        public string Image { get; set; }
        [JsonProperty("current_price")]
        public decimal Current_price { get; set; }
        [JsonProperty("market_cap", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Market_cap { get; set; }
        [JsonProperty("market_cap_rank", NullValueHandling = NullValueHandling.Ignore)]
        public int Market_cap_rank { get; set; }
        [JsonProperty("fully_diluted_valuation", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Fully_diluted_valuation { get; set; }
        [JsonProperty("total_volume", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Total_volume { get; set; }
        [JsonProperty("high_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal High_24h { get; set; }
        [JsonProperty("low_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Low_24h { get; set; }
        [JsonProperty("price_change_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Price_change_24h { get; set; }
        [JsonProperty("price_change_percentage_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Price_change_percentage_24h { get; set; }
        [JsonProperty("market_cap_change_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Market_cap_change_24h { get; set; }
        [JsonProperty("market_cap_change_percentage_24h", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Market_cap_change_percentage_24h { get; set; }
        [JsonProperty("circulating_supply", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Circulating_supply { get; set; }
        [JsonProperty("total_supply", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Total_supply { get; set; }
        [JsonProperty("max_supply", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Max_supply { get; set; }
        [JsonProperty("ath", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Ath { get; set; }
        [JsonProperty("ath_change_percentage", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Ath_change_percentage { get; set; }
        [JsonProperty("ath_date")]
        public DateTime Ath_date { get; set; }
        [JsonProperty("atl", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Atl { get; set; }
        [JsonProperty("atl_change_percentage", NullValueHandling = NullValueHandling.Ignore)]
        public decimal Atl_change_percentage { get; set; }
        [JsonProperty("atl_date")]
        public DateTime Atl_date { get; set; }
        [JsonProperty("roi")]
        public object Roi { get; set; }
        [JsonProperty("last_updated")]
        public DateTime Last_updated { get; set; }

    }
}